be/src/util/brpc_stub_cache.h

```diff
@@ -34,25 +34,32 @@ class BrpcStubCache {
     BrpcStubCache();
     virtual ~BrpcStubCache();
 
-    virtual PBackendService_Stub* get_stub(const butil::EndPoint& endpoint) {
-        std::lock_guard<SpinLock> l(_lock);
+    virtual std::shared_ptr<PBackendService_Stub> get_stub(const butil::EndPoint& endpoint) {
+        std::lock_guard<std::mutex> l(_mutex);
         auto stub_ptr = _stub_map.seek(endpoint);
         if (stub_ptr != nullptr) {
-            return *stub_ptr;
+            if (available(*stub_ptr, endpoint)) {
```

Comment:
It is not good to check it every time we call `get_stub`. I think it should be checked when error happens.

Reply:
Checking availability is a low-cost check. Checking and refreshing before obtaining the stub is more appropriate than handling it after failure.